### PR TITLE
LWDEV-8765 LimitOrderCancelService/LimitOrderMassCancelService use context

### PR DESCRIPTION
**Pull request by rocketmannot**

### CODE REVIEW

File: src/main/kotlin/com/lykke/matching/engine/incoming/MessageRouter.kt [hunk 1]
```diff
@@ -8,17 +8,19 @@ import java.util.concurrent.BlockingQueue
...
 ) {
     fun process(wrapper: MessageWrapper) {
-        when(wrapper.type) {
-            MessageType.CASH_IN_OUT_OPERATION.type -> cashInOutQueue.put(wrapper)
-            MessageType.CASH_TRANSFER_OPERATION.type -> cashTransferQueue.put(wrapper)
+        when (wrapper.type) {
+            MessageType.CASH_IN_OUT_OPERATION.type -> cashInOutInputQueue.put(wrapper)
+            MessageType.CASH_TRANSFER_OPERATION.type -> cashTransferInputQueue.put(wrapper)
+            MessageType.LIMIT_ORDER_CANCEL.type,
+            MessageType.OLD_LIMIT_ORDER_CANCEL.type -> limitOrderCancelInputQueue.put(wrapper)
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/incoming/MessageRouter.kt]: Add section for MassCancelOperation and limitOrderMassCancelInputQueue.

File: src/main/kotlin/com/lykke/matching/engine/incoming/parsers/impl/LimitOrderMassCancelOperationContextParser.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        messageWrapper.messageId = messageId
+        messageWrapper.id = message.uid
+        messageWrapper.timestamp = Date().time
+
+        val messageType =  MessageType.valueOf(messageWrapper.type) ?: throw Exception("Unknown message type ${messageWrapper.type}")
+
+        val assetPairId = if (message.hasAssetPairId()) message.assetPairId else null
+        val isBuy = if (message.hasIsBuy()) message.isBuy else null
+
+        return LimitOrderMassCancelOperationContext(message.uid, message.messageId, message.clientId,
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/incoming/parsers/impl/LimitOrderMassCancelOperationContextParser.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/services/LimitOrderCancelService.kt [hunk 3]
```diff
@@ -10,123 +12,94 @@ import com.lykke.matching.engine.messages.MessageType
...
-        if (orders.isEmpty() && stopOrders.isEmpty()) {
-            LOGGER.info("Unable to find order ids: $notFoundOrderIds")
-            writeResponse(messageWrapper, MessageStatus.LIMIT_ORDER_NOT_FOUND)
-            return Orders.processed()
-        }
+        val updateSuccessful = limitOrdersCancelHelper.cancelOrders(LimitOrdersCancelHelper.CancelRequest(context.uid,
+                context.messageId,
+                context.messageType,
+                typeToOrder[LimitOrderType.LIMIT],
+                typeToOrder[LimitOrderType.STOP_LIMIT], now, context.processedMessage, false))
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/services/LimitOrderCancelService.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/services/LimitOrderMassCancelService.kt [hunk 4]
```diff
@@ -1,47 +1,54 @@
...
+
+        val updateSuccessful = limitOrdersCancelHelper.cancelOrders(LimitOrdersCancelHelper.CancelRequest(context.uid,
+                context.messageId,
+                context.messageType,
+                getOrders(context),
+                getStopOrders(context),
+                now,
+                context.processedMessage,
+                false))
+
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/services/LimitOrderMassCancelService.kt]: fixed

**Pull request merged**